apps/rush-lib/src/api/CommandLineConfiguration.ts

```diff
@@ -7,20 +7,87 @@ import { JsonFile, JsonSchema, FileSystem } from '@rushstack/node-core-library';
 
 import { RushConstants } from '../logic/RushConstants';
 
-import {
+import type {
   CommandJson,
   ICommandLineJson,
   IPhaseJson,
-  ParameterJson,
-  IPhasedCommandJson
+  IPhasedCommandJson,
+  IBulkCommandJson,
+  IGlobalCommandJson,
+  IFlagParameterJson,
+  IChoiceParameterJson,
+  IStringParameterJson
 } from './CommandLineJson';
+import type { RushConfigurationProject } from './RushConfigurationProject';
 
 const EXPECTED_PHASE_NAME_PREFIX: '_phase:' = '_phase:';
 
 export interface IShellCommandTokenContext {
   packageFolder: string;
 }
 
+export interface IPhase extends IPhaseJson {
+  isSynthetic?: boolean;
+  logFilenameIdentifier: string;
+  getDisplayNameForProject(rushProject: RushConfigurationProject): string;
```

Comment:
`IPhase` seems to be a code object with member functions.  It's odd that it extends from `IPhaseJson` which is meant to be a JSON-serializable format.

Reply:
Removed `getDisplayNameForProject`.